### PR TITLE
Dispatch

### PR DESCRIPTION
**Pull request by ArtOfCode-**
2nd attempt. CAD only.

### CODE REVIEW

**Comment by izwick-schachter** [comment]
I read through the code and it looked good. About to test locally and look at the changes there.

**Comment by izwick-schachter** [comment]
Looks good now. Merge if you want.

File: app/controllers/rescue_requests_controller.rb [hunk 1]
```diff
@@ -36,14 +36,21 @@ def disaster_index
...
+  end
 
   def create
-    @request = @disaster.rescue_requests.new(lat: params[:lat], long: params[:long], key: SecureRandom.hex(32))
+    medical_conditions = params.permit(MedicalCondition.all.map { |m| "conditions_#{m.id}".to_sym }).to_hash
+                               .map { |key| MedicalCondition.find(key.to_s.split('_').last.to_i) }
+    @request = @disaster.rescue_requests.new request_params.merge(key: SecureRandom.hex(32), request_status: RequestStatus['New'], request_priority: RequestPriority['New'], medical_conditions: medical_conditions)
     if @request.save
-      render json: { status: 'success', id: @request.id, key: @request.key }
+      puts "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\n\n\n\n\n\n\n\n"
```
Comment (ArtOfCode-) [inline app/controllers/rescue_requests_controller.rb]: uh

File: app/controllers/rescue_requests_controller.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@ def index
   end
 
   def disaster_index
-    status_ids = [RequestStatus.find_by(name: 'Rescued').id, RequestStatus.find_by(name: 'Closed').id]
+    status_ids = [RequestStatus.find_by(name: 'Safe').id, RequestStatus.find_by(name: 'Closed').id]
```
Comment (ArtOfCode-) [inline app/controllers/rescue_requests_controller.rb]: These can use the `[]` syntax to make them shorter: `RequestStatus['Safe'], RequestStatus['Closed']`.

File: db/seeds.rb [hunk 3]
```diff
@@ -7,6 +7,10 @@
 Dir.glob(Rails.root.join('db/seeds/**/*.yml')).each do |f|
   basename = Pathname.new(f).relative_path_from(Pathname.new(Rails.root.join('db/seeds'))).to_s
   type = basename.gsub('.yml', '').singularize.classify.constantize
-  data = YAML.load_file f
-  type.create data
+  begin
+    data = YAML.load_file f
+    type.create data
+  rescue StandardError => e
+    puts "Got error #{e}. Continuing..."
```
Comment (ArtOfCode-) [inline db/seeds.rb]: What error were you getting that made this necessary? It's good to have it, but if there's a bug we should also fix that.

File: app/controllers/rescue_requests_controller.rb (hunk 1 above)
Comment (izwick-schachter) [reply app/controllers/rescue_requests_controller.rb]: Whoops. Debugging.

File: db/seeds.rb (hunk 3 above)
Comment (ArtOfCode-) [reply db/seeds.rb]: Huh?

**Comment by izwick-schachter** [comment]
@ArtOfCode- Wanna put this into the navbar before we merge?

**Comment by ArtOfCode-** [comment]
I'll do it on master :)

**Pull request merged**